js/lessonfilter.js

```diff
@@ -60,51 +127,128 @@ function wireButtons() {
   console.log(uri.toString());
 
   var options = {
-    valueNames: [ 'date', 'title', 'difficulty', 'activity', 'topics' ]
+    valueNames: [ 'date', 'title', 'difficulty', 'activity', 'topics','abstract', 'content' ]
   };
 
   var featureList = new List('lesson-list', options);
   // We need a stateObj for adjusting the URIs on button clicks, but the value is moot for now; could be useful for future functionality.
   var stateObj = { foo: "bar" };
 
+  // Get search indices and corpuses using URI to make a call to search-index for each language
+  let idx;
+  let corpus;
+
+  // Load search data
+  function loadSearchData() {
+    // Get language specific corpus and index
+    const language = uri.toString().split('/').slice(-3)[0];
+    $.getJSON(`https://programminghistorian.github.io/search-index/indices/index${language.toUpperCase()}.json`).done(response => {
+      idx = lunr.Index.load(JSON.parse(JSON.stringify(response)));
+    });
+    $.getJSON(`https://programminghistorian.org/${language}/search.json`).done(response => {
+      corpus = response;
+      // Enable search input and button once data loaded
+      $('#search').attr("placeholder", "Type search terms...");
```

Comment:
Since we can't (easily) reference the translation snippets YAML from javascript, is there a way you could add a hidden element to lesson-index.html where we CAN load in `{{ site.data.snippets.type-search-terms[page.lang] }}` and then grab its value to insert as the placeholder here? Or some other way so that we don't have to hardcode display language in the JS?

Reply:
That sounds like a great solution! I'll pull down your updates and try it out :)

Reply:
(I'm 95% sure this is used on some other part of the site filter JS haha and I'm just lifting the idea)

Reply:
Lolz... I ended up just doing it through `lesson-index.html` and `style.css`, where we just hide and show from inputs from `lessonfilter`. It seemed less complicated but feel free to do any edits!